### PR TITLE
Putting basic authentication and authorization code in place

### PR DESCRIPTION
**Pull request by amitxagarwal**
Putting basic authentication and authorization code in place

### CODE REVIEW

File: Kmd.Momentum.Mea.Common/Authorization/HasResourceHandler.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using System.Linq;
```
Comment (Ashish0410verma) [inline Kmd.Momentum.Mea.Common/Authorization/HasResourceHandler.cs]: Please remove unused using blocks.

File: Kmd.Momentum.Mea.Common/Authorization/HasResourceRequirement.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
```
Comment (Ashish0410verma) [inline Kmd.Momentum.Mea.Common/Authorization/HasResourceRequirement.cs]: We can get rid of unused using blocks.

File: src/Kmd.Momentum.Mea.Api/Startup.cs [hunk 3]
```diff
@@ -35,6 +39,16 @@ public void ConfigureServices(IServiceCollection services)
...
             services.AddControllers();
 
+            services.AddAuthentication(AzureADB2CDefaults.BearerAuthenticationScheme)
+                .AddAzureADB2CBearer(options => Configuration.Bind("AzureAdB2C", options));
+
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy(Resource.Access, policy => policy.Requirements.Add(new HasResourceRequirement(Resource.Access)));
+            });
+
```
Comment (ExploreMyCode) [inline src/Kmd.Momentum.Mea.Api/Startup.cs]: Make sure to test based on different resource access.

Comment (amitxagarwal) [reply src/Kmd.Momentum.Mea.Api/Startup.cs]: This cannot be tested as of now as this is just the skeleton and we dont have this api registered with Logic Azure ad. There is more code to come here.

**Comment by Ashish0410verma** [comment]
Reviewed and skeleton looks OK to me

**Comment by ExploreMyCode** [comment]
Looks good to me.

**Pull request closed without merging**